walkthroughs/howto-k8s-connection-pools/v1beta2/manifest.yaml.template

```diff
@@ -0,0 +1,298 @@
+  awsName: color-paths.${APP_NAMESPACE}.svc.cluster.local
+  provider:
+    virtualRouter:
+      virtualRouterRef:
+        name: color-paths
+---
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualRouter
+metadata:
+  namespace: ${APP_NAMESPACE}
+  name: color-paths
+spec:
+  listeners:
+    - portMapping:
+        port: 8080
+        protocol: http
+  routes:
+    - name: color-route-green
+      priority: 20
+      httpRoute:
+        match:
+          prefix: /green
+        action:
+          weightedTargets:
+            - virtualNodeRef:
+                name: green
+              weight: 1
+    - name: color-route-red
+      priority: 30
+      httpRoute:
+        match:
+          prefix: /red
+        action:
+          weightedTargets:
+            - virtualNodeRef:
+                name: red
+              weight: 1
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: color-green
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 8080
+      name: http
+  selector:
+    app: color
+    version: green
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: green
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: color
+      version: green
+  template:
+    metadata:
+      labels:
+        app: color
+        version: green
+    spec:
+      containers:
+        - name: app
+          imagePullPolicy: Always
+          image: ${COLOR_APP_IMAGE}
+          ports:
+            - containerPort: 8080
+          env:
+            - name: "COLOR"
+              value: "green"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: color-red
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 8080
+      name: http
+  selector:
+    app: color
+    version: red
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: red
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: color
+      version: red
+  template:
+    metadata:
+      labels:
+        app: color
+        version: red
+    spec:
+      containers:
+        - name: app
+          imagePullPolicy: Always
+          image: ${COLOR_APP_IMAGE}
+          ports:
+            - containerPort: 8080
+          env:
+            - name: "COLOR"
+              value: "red"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: color-paths
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 8080
+      name: http
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ingress-gw
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 80
+      targetPort: 8088
+      name: http
+  selector:
+    app: ingress-gw
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ingress-gw
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ingress-gw
+  template:
+    metadata:
+      labels:
+        app: ingress-gw
+    spec:
+      containers:
+        - name: envoy
+          image: controller-override
```

Comment:
@fawadkhaliq how does this work? where does this image come from?

Reply:
okay, found it https://github.com/aws/aws-app-mesh-controller-for-k8s/pull/262/files#diff-67827a4affa2ad0d56ada1256f5cb2138356b1b5f5ff47afab7ca15cf3d28444R19

Reply:
that fails for me with an error in the  `spec.containers[1].ports[0].containerPort:` even tho I have a single container only...

Reply:
it seems this only happens when x-ray is enabled for the cluster...